spec/controllers/categories_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.describe CategoriesController, type: :controller do
+  let(:good_params) do
+    {category: {
+      name:"category"}
+    }
+  end
+  
+  let(:bad_params) do
+    {category: {
+      name: nil
+      }}
+  end
+
+  let(:merchant) do
+    Merchant.create(username: "Apple", email: "email345@email.com", password: "password", password_confirmation: "password")
+  end
+
+  let(:session_data) do
+    {
+      username: "Apple",
+      password: "password"
+    }
+  end
+
+  describe "GET 'new'" do
+    it "is successful" do
+      merchant.authenticate(session_data)
+      session[:user_id] = merchant.id
+
+      get :new
+      expect(response.status).to eq 200
+      expect(subject).to render_template :new
+    end
+  end
+
+  describe "POST 'create'" do
+    it "successful create redirects to category show page" do
+      expect(Category.all.length).to eq 0
+      post :create, good_params
+      expect(response.status).to eq 302
+      expect(Category.all.length).to eq 1
```

Comment:
You could use `expect{ action }.to change(Category.count)` here! See second note on line 164 of spec/controllers/products_controller_spec.rb.